R/nm-tables.R

```diff
@@ -84,14 +84,12 @@ nm_tables <- function(
 #' @export
 nm_table_files <- function(.mod, .check_exists = TRUE) {
   .p <- get_model_path(.mod)
-  .l <- parse_ctl_to_list(.p)
+  .l <- nmrec::read_ctl(.p)
   out_dir <- get_output_dir(.mod, .check_exists = .check_exists)
 
   # get file names from table statements and construct paths
-  .f <- .l[grep("^TAB", names(.l))] %>%
-    map_chr(~paste(.x, collapse = " ")) %>%
-    str_extract("\\bFILE\\s*=\\s*([^ ]+)") %>%
-    str_replace("\\bFILE\\s*=\\s*", "") %>%
+  .f <- nmrec::select_records(.l, "table") %>%
+    purrr::map_chr(~nmrec::get_record_option(.x, "file")$value) %>%
     str_replace("^\\.\\/", "") %>%
     file.path(out_dir, .)
 
```

Comment:
This isn't an issue introduced by the changes proposed here, but I think it'd be okay to address with this PR (up to you): the file value may be quoted with single or double quotes. Could append something like this to end of pipeline:

```
> .f
[1] "foo"     "'bar'"   "\"baz\""
> sub("^'(.*)'$", "\\1", .f) %>% sub('^"(.*)"$', "\\1", .)
[1] "foo" "bar" "baz"
```

Reply:
added - tested that this worked well too